groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java

```diff
@@ -127,6 +145,59 @@ public Path getWorkspaceRoot() {
     }
 
     @Override
+    public void handleFileChanged(Path originalFile, List<TextDocumentContentChangeEvent> contentChanges) {
+        try {
+            SourceWriter sourceWriter = null;
+            if (originalSourceToChangedSource.containsKey(originalFile)) {
+                // New change on existing changed source
+                sourceWriter = originalSourceToChangedSource.get(originalFile);
+            } else {
+                // New source to switch out
+                Path newSourcePath = changedFilesRoot.resolve(workspaceRoot.relativize(originalFile));
+                sourceWriter = SourceWriter.of(originalFile, newSourcePath);
+                originalSourceToChangedSource.put(originalFile, sourceWriter);
+            }
+            // Apply changes to source writer and reset compilation unit
+            sourceWriter.applyChanges(contentChanges);
+            resetCompilationUnit();
+        } catch (IOException e) {
+            Throwables.propagate(e);
+        }
+    }
+
+    @Override
+    public void handleFileClosed(Path originalFile) {
+        if (!originalSourceToChangedSource.containsKey(originalFile)) {
+            return;
+        }
+        Path changedSource = originalSourceToChangedSource.get(originalFile).getDestination();
+        originalSourceToChangedSource.remove(originalFile);
+        // Deleted the changed file
+        if (!changedSource.toFile().delete()) {
+            throw new IllegalStateException(String.format("Unable to delete file %s", changedSource.toAbsolutePath()));
```

Comment:
actually I think I was wrong... unable to delete this file shouldn't be too bad since we after we remove the `originalFile ` key from the map, it shouldn't try to add the changed file to the compilationUnit anymore. So logging the error should be enough